compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenAttrs.h

```diff
@@ -122,11 +125,17 @@ inline LogicalResult setOpConfigAndEntryPointFnTranslation(
     IREE::Codegen::DispatchLoweringPassPipeline passPipeline,
     ArrayRef<int64_t> workgroupSize = {},
     std::optional<int64_t> subgroupSize = {},
-    DictionaryAttr pipelineConfig = DictionaryAttr()) {
+    DictionaryAttr pipelineConfig = DictionaryAttr(),
+    std::optional<StringAttr> rootOpInfo = std::nullopt) {
   MLIRContext *context = entryPointFn.getContext();
   auto config = IREE::Codegen::LoweringConfigAttr::get(context, tileSizes,
                                                        scalableTileFlags);
   setLoweringConfig(op, config);
+  rootOpInfo = mlir::StringAttr::get(
+      context, "this-is-root-op"); // TODO: add more info for tuner
+  if (rootOpInfo.has_value()) {
+    setRootOpInfo(op, rootOpInfo.value());
```

Comment:
Should we make it a UnitAttribute until we have any useful information to encode?

Also, this seems backwards to me -- we set the `rootOpInfo` in the line above so it should always have a value

Reply:
Dont see why not just make this a `UnitAttribute` now. Also `this-is-root-op` is too verbose in IR.